### PR TITLE
added delete_entry, delete_group

### PR DESCRIPTION
**Pull request by Evidlo**
Not really much has changed.  My last PR just had a bunch of tiny commits.

- added delete_entry, delete_group functions
- added appropriate tests and documentation

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -3,8 +3,6 @@ pykeepass
 
 This library allows you to write entries to a KeePass database
 
-*pykeepass does not currently support Python 3*
-
```
Comment (pschmitt) [inline README.rst]: Did you intend to remove this?

File: pykeepass/pykeepass.py [hunk 2]
```diff
@@ -9,13 +9,12 @@
 import libkeepass
 import logging
 import os
-import re
```
Comment (pschmitt) [inline pykeepass/pykeepass.py]: re is actually used in pykeepass...

File: pykeepass/pykeepass.py [hunk 3]
```diff
@@ -77,8 +76,7 @@ def __xpath(self, xpath_str, tree=None):
 
     #---------- Groups ----------
 
-    def find_groups_by_name(self, group_name, tree=None, regex=False,
-                            first=False):
+    def find_groups_by_name(self, group_name, tree=None, regex=False, first=False):
```
Comment (pschmitt) [inline pykeepass/pykeepass.py]: 80+ chars

File: pykeepass/pykeepass.py [hunk 4]
```diff
@@ -91,10 +89,8 @@ def find_groups_by_name(self, group_name, tree=None, regex=False,
 
         return res
 
-    def find_groups_by_path(self, group_path_str=None, regex=False, tree=None,
-                            first=False):
-        logger.info('Looking for group {}'.format(
-            group_path_str if group_path_str else 'Root'))
+    def find_groups_by_path(self, group_path_str=None, regex=False, tree=None, first=False):
+        logger.info('Looking for group {}'.format(group_path_str if group_path_str else 'Root'))
```
Comment (pschmitt) [inline pykeepass/pykeepass.py]: 80+ chars

File: pykeepass/pykeepass.py [hunk 5]
```diff
@@ -123,10 +119,12 @@ def add_group(self, destination_group, group_name):
...
         return group
 
+    def delete_group(self, group):
+        group.delete()
+
     #---------- Entries ----------
 
-    def __find_entry_by(self, key, value, regex=False, tree=None,
-                        history=False, first=False):
+    def __find_entry_by(self, key, value, regex=False, tree=None, history=False, first=False):
```
Comment (pschmitt) [inline pykeepass/pykeepass.py]: 80+ chars

**Comment by Evidlo** [comment]
Anything else you'd like me to do?

**Pull request merged**